components/ClickAwayListener/ClickAwayListener.tsx

```diff
@@ -0,0 +1,20 @@
+import React, { FC, ReactElement, RefObject, useRef } from 'react';
+import useOnClickAway from 'Hooks/useOnClickAway';
+
+type Props = {
+  onClickAway: () => void;
+};
+
+const ClickAwayListener: FC<Props> = ({ children, onClickAway }) => {
+  const ref = useRef<HTMLElement>(null);
+  useOnClickAway(ref, () => onClickAway());
+
+  if (React.Children.count(children) == 1) {
+    const copyWithRef = React.cloneElement(children as ReactElement, { ref: ref });
+    return <React.Fragment>{copyWithRef}</React.Fragment>;
```

Comment:
You can just use `<></>` as a shorthand of `<React.Fragment></React.Fragment>` 🙂

Reply:
Thats true. It stays in this form after i tried different approaches including passing some props to React.Fragment. I will change that ✌